src/OVHSwiftAdapter.php

```diff
@@ -127,6 +127,48 @@ public function getTemporaryUrl(string $path, \DateTimeInterface $expiresAt, arr
         );
     }
 
+    public function getFormPostSignature(string $path, array $options = []): string
+    {
+        // Ensure Temp URL Key is provided for the Disk
+        if (empty($this->config->getTempUrlKey())) {
+            throw new \InvalidArgumentException("No Temp URL Key provided for container '".$this->container->name."'");
+        }
+
+        // expires is required otherwise a valid request will not be possible to be formed
+        if (!isset($options['expires'])) {
+            throw new \InvalidArgumentException("Missing 'expires' key in options");
+        }
+
+        // check if the 'expires' values is in the past
+        if (($options['expires'] ?? 0) < time()) {
+            throw new \InvalidArgumentException("Value of 'expires' cannot be in the past");
+        }
+
+        // Ensure $path doesn't begin with a slash
+        $path = ltrim($path, '/');
+
+        // Get the method
+        $method = 'POST';
+        $redirectUrl = $options['redirect'] ?? '';
+        $maxFileSize = $options['max_file_size'] ?? 25 * 1024 * 1024;
+        $maxFileCount = $options['max_file_count'] ?? 1;
+        $linkExpiry = $options['expires'];
+
+        // The url on the OVH host
+        $codePath = sprintf(
+            '/v1/AUTH_%s/%s/%s',
+            $this->config->getProjectId(),
+            $this->config->getContainerName(),
+            $path
+        );
+
+        // Body for the HMAC hash
+        $body = sprintf("%s\n%s\n%s\n%s\n%s", $codePath, $redirectUrl, $maxFileSize, $maxFileCount, $linkExpiry);
+
+        // The actual hash signature
+        return hash_hmac('sha1', $body, $this->config->getTempUrlKey());
```

Comment:
Maybe we could abstract this logic with the TempUrlKey signature generation?

Reply:
Agree. Let me get to doing some changes